fail2ban/helpers.py

```diff
@@ -144,6 +158,36 @@ def splitwords(s):
 	return filter(bool, map(str.strip, re.split('[ ,\n]+', s)))
 
 
+#
+# Following "uni_decode" function unified python independent any to string converting
+#
+# Typical example resp. work-case for understanding the coding/decoding issues:
+#
+#   [isinstance('', str), isinstance(b'', str), isinstance(u'', str)]
+#   [True, True, False]; # -- python2
+#	  [True, False, True]; # -- python3
+#
+if sys.version_info >= (3,):
+	def uni_decode(x, enc=PREFER_ENC, errors='strict'):
+		try:
+			if isinstance(x, bytes):
+				return x.decode(enc, errors)
+			return x
+		except (UnicodeDecodeError, UnicodeEncodeError): # pragma: no cover - unsure if reachable
+			if errors != 'strict': 
```

Comment:
Nope, otherwise it would always throw an error (this is test covered also). 
The normal case is to execute it without parameter (so trying strict), then the worst-case:
`uni_decode(str)` produces an error with strict, but does not raise an error (`errors != 'strict'`) and try convert with replace, if it also produces an error, this will be reraised...